File: 8 kyu/Floating point comparison/solution.py
def approx_equals(a, b):
    return abs(a-b) < 0.001
  
###############
from math import isclose

def approx_equals(a, b):
    return isclose(a, b, rel_tol=0, abs_tol=1e-3)
  
############
def approx_equals(a, b):
    return abs(a - b) < 1e-3
  
#############
from math import isclose

def approx_equals(a, b):
    return isclose(a, b, abs_tol=1e-3)
  
##############
def approx_equals(a, b):
    a = round(a, 5)
    b = round(b, 5)
    return abs(a - b) <= 0.001
  
###############
import math

def approx_equals(a, b):
    a = round(a, 5)
    b = round(b, 5)
    l = math.isclose(a, b, abs_tol = 0.001)
    return l
#     if a - b > rel_tol:
#         a != b
#     if a != b:
#         return False
#     else:
#         return True
#############################
def approx_equals(a, b):
    if a == b or (a != b and abs(a - b) <= 0.001):
        return True
    else:
        return False
      
####################
def approx_equals(a, b):
    c = abs(a-b)
    print(c)
    c = abs(a-b) <= 0.001
    return c

File: 8 kyu/Floating point comparison/solution.cpp
#include <cmath>
using namespace std;

bool approx_equals(double a, double b) {
  return abs(a - b) < 0.001;
}

###################
#include <cmath>
using namespace std;

bool approx_equals(double a, double b) {
  return abs(a - b) <= 1e-3;
}

#######################
#include <cmath>
using namespace std;

bool approx_equals(double a, double b, float epsilon = 0.001f) {
  return fabs(a-b) < epsilon;
  //return round(a*1000) == round(b*1000);
}

#####################
#include <cmath>
using namespace std;

bool approx_equals(double a, double b) {
  float diff;
  
  if (a>=b){
  diff = a-b;
    }
  
  if (b>a){
    diff = b-a;
  }
  
  if (diff <= 0.001){
    return true;
  }
  else {
      return false;
}
}

######################
#include <cmath>
using namespace std;

bool approx_equals(double a, double b) {
  if(abs(a - b) < 0.001){
    return true; 
  }
  return false; 
}

################
#include <cmath>
using namespace std;

bool approx_equals(double a, double b) {
//   return trunc(a*1000) == trunc(b*1000);
  return abs(a - b) < 0.001;
}
